### PR TITLE
fix(rn): get login working e2e

### PR DESCRIPTION
**Pull request by moldy530**
# Pull Request Checklist

- [ ] Did you add new tests and confirm existing tests pass? (`yarn test`)
- [ ] Did you update relevant docs? (docs are found in the `site` folder, and guidelines for updating/adding docs can be found in the [contribution guide](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md))
- [ ] Do your commits follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] Does your PR title also follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) standard?
- [ ] If you have a breaking change, is it [correctly reflected in your commit message](https://www.conventionalcommits.org/en/v1.0.0/#examples)? (e.g. `feat!: breaking change`)
- [ ] Did you run lint (`yarn lint:check`) and fix any issues? (`yarn lint:write`)
- [ ] Did you follow the [contribution guidelines](https://github.com/alchemyplatform/aa-sdk/blob/main/CONTRIBUTING.md)?

---

## PR-Codex overview
This PR focuses on enhancing the `NativeTEKStamper` integration within the `React Native` application, improving module exports, and updating dependencies.

### Detailed summary
- Changed export of `NativeTEKStamper` from default to named export.
- Updated Android module configuration for `isTurboModule`.
- Added dotenv configuration in `build.gradle`.
- Updated `package.json` with new dependencies.
- Enhanced `SignerClient` class with new methods for authentication.
- Improved `NativeTEKStamperModule` with HPKE decryption support.
- Updated various dependencies in `yarn.lock` for better compatibility.

> The following files were skipped due to too many changes: `account-kit/rn-signer/example/yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by moldy530** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1132?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#1132** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1132?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#1131** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1131?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1130** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1130?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1129** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1129?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#1128** <a href="https://app.graphite.dev/github/pr/alchemyplatform/aa-sdk/1128?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @moldy530 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt [hunk 1]
```diff
@@ -75,9 +86,21 @@ class NativeTEKStamperModule(reactContext: ReactApplicationContext) :
...
+        .setVariant(HpkeParameters.Variant.NO_PREFIX)
+        .build()
+
     init {
         TinkConfig.register()
 
+        if (Security.getProvider(BouncyCastleProvider.PROVIDER_NAME).javaClass != BouncyCastleProvider::class.java) {
+            Security.removeProvider(BouncyCastleProvider.PROVIDER_NAME)
+        }
+
```
Comment (moldy530) [inline account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt:103]: This was due to the fact that android adds its own bouncy castle implementation that doesn't include `EC`

File: account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt [hunk 2]
```diff
@@ -135,38 +155,58 @@ class NativeTEKStamperModule(reactContext: ReactApplicationContext) :
...
                 .putString(
                     BUNDLE_PUBLIC_KEY,
-                    DatatypeConverter.printHexBinary(publicKeyBytes).uppercase()
+                    DatatypeConverter.printHexBinary(publicKeyBytes).lowercase()
                 )
                 .apply()
 
             return promise.resolve(true)
         } catch (e: Exception) {
+            Log.e("error", "an error happened", e)
```
Comment (moldy530) [inline account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt:209]: oops, remove this

File: account-kit/rn-signer/example/src/App.tsx [hunk 3]
```diff
@@ -1,9 +1,69 @@
-import { StyleSheet, Text, View } from "react-native";
+/* eslint-disable import/extensions */
+import { NativeTEKStamper } from "@account-kit/react-native-signer";
+import type { User } from "@account-kit/signer";
+import { useEffect, useState } from "react";
+import { Button, StyleSheet, Text, TextInput, View } from "react-native";
+import { SignerClient } from "./signer";
 
 export default function App() {
+  const [publicKey, setPublicKey] = useState<string | null>(null);
```
Comment (moldy530) [inline account-kit/rn-signer/example/src/App.tsx:9]: ⚠️ **[eslint]** <[@typescript-eslint/no-unused-vars](https://typescript-eslint.io/rules/no-unused-vars)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>'publicKey' is assigned a value but never used.

File: account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt [hunk 4]
```diff
@@ -234,22 +275,56 @@ class NativeTEKStamperModule(reactContext: ReactApplicationContext) :
...
+
+    private fun getHpkePrivateKeyFromKeysetHandle(keysetHandle: KeysetHandle): HpkePrivateKey {
+        val pkKs = CleartextKeysetHandle.getKeyset(keysetHandle)
+        val pkKeyData = pkKs.keyList[0].keyData
+        if (pkKeyData.typeUrl != "type.googleapis.com/google.crypto.tink.HpkePrivateKey") {
+            throw Error("invalid key type")
+        }
+
+        return HpkePrivateKey.create(
+            com.google.crypto.tink.hybrid.HpkePublicKey.create(
```
Comment (moldy530) [inline account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt:309]: we use the proto version of the public key elsewhere hence the need for this specific one. I'm going to look into refactoring things so we can use the same interfaces throughout

File: account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt [hunk 5]
```diff
@@ -234,22 +275,56 @@ class NativeTEKStamperModule(reactContext: ReactApplicationContext) :
...
+            ),
+            SecretBytes.copyFrom(
+                com.google.crypto.tink.proto.HpkePrivateKey.parseFrom(pkKeyData.value).privateKey.toByteArray(),
+                InsecureSecretKeyAccess.get()
             )
         )
-        return DatatypeConverter.printHexBinary(outputStream.toByteArray()).uppercase()
+    }
+
+    private fun convertToUncompressedPublicKeyBytes(ephemeralPublicKey: ECPublicKey): ByteArray {
```
Comment (moldy530) [inline account-kit/rn-signer/android/src/main/java/com/accountkit/reactnativesigner/NativeTEKStamperModule.kt:321]: I'm gonna follow up and actually make this a helper that takes in: `key: bytes, inputFormat: PointFormatType, outputFormat: PointFormatType` to make things a bit cleaner

File: account-kit/rn-signer/example/src/signer.ts
Comment (moldy530) [inline account-kit/rn-signer/example/src/signer.ts:1]: This is a super barebones signer implementation that needs to move out of the example and in to the RN src, have this here to test

**Comment by moldy530** [comment]
Was able to:
- generate a link with a bundler
- copy the bundle in
- complete email auth!

(bundle omitted from image, the user: ... is what we returned from our servers on successful whoami request post login)

File: account-kit/rn-signer/src/index.tsx [hunk 6]
```diff
@@ -1,3 +1,4 @@
-import NativeTEKStamper from "./NativeTEKStamper.js";
+// eslint-disable-next-line import/extensions
+import NativeTEKStamper from "./NativeTEKStamper";
 
-export default NativeTEKStamper;
+export { NativeTEKStamper };
```
Comment (dphilipson) [inline account-kit/rn-signer/src/index.tsx:4]: Can this be
```ts
export { NativeTEKStamper } from "./NativeTEKStamper";
```
?

**Review by dphilipson — APPROVED** [review]
Holy cow this is intense. Great work!

**Comment by iykazrji** [comment]
Just going through this! This is glorious @moldy530 ! 🔥🔥🔥

**Pull request merged**